### PR TITLE
Various fixes for tee-supplicant filesystem

### PR DESCRIPTION
**Pull request by jbech-linaro**
I started fixing a buffer overrun, but then I continued cleaning up a bit more.

### CODE REVIEW

File: tee-supplicant/src/tee_supp_fs.c [hunk 1]
```diff
@@ -188,120 +188,160 @@ static int tee_supp_fs_conv_accessflags(int in)
 	return flags;
 }
 
-static size_t tee_supp_fs_create_fname(char *file, char *out)
+static size_t tfs_get_absolute_filename(char *file, char *out, size_t out_size)
```
Comment (jbech-linaro) [inline tee-supplicant/src/tee_supp_fs.c]: The comments added [here](https://github.com/jbech-linaro/optee_client/commit/f25aa194f4afc11b1af91e67ad20728742385051#commitcomment-10909714) (before I pushed using force) has been taken into account.

**Comment by jenswi-linaro** [comment]
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jbech-linaro** [comment]
All comments has been taken into account and tags have been applied.

**Pull request merged**